lib/src/impl_ffi/impl_ffi.random.dart

```diff
@@ -30,4 +30,28 @@ final class _RandomImpl implements RandomImpl {
       dest.setAll(0, out.asTypedList(dest.length));
     });
   }
+
+  @override
+  String randomUUID() {
+    return _Scope.sync((scope) {
+      final out = scope<ffi.Uint8>(16);
+      _checkOp(ssl.RAND_bytes(out, 16) == 1);
+
+      var bytes = out.asTypedList(16);
+
+      // Set the 4 most significant bits of bytes[6] to 0100.
+      bytes[6] = (bytes[6] & 0x0F) | 0x40;
+
+      // Set the 2 most significant bits of bytes[8] to 10.
+      bytes[8] = (bytes[8] & 0x3F) | 0x80;
+
+      return bytes
+          .map((e) => e.toRadixString(16).padLeft(2, '0'))
+          .join()
+          .replaceAllMapped(
+            RegExp(r'(\w{8})(\w{4})(\w{4})(\w{4})(\w{12})'),
+            (m) => '${m[1]}-${m[2]}-${m[3]}-${m[4]}-${m[5]}',
+          );
```

Comment:
I've done this other places the classical implementation is:
```
final bytes = _randomBytes(16);
  // Set V4 bits according to:
  // https://tools.ietf.org/html/rfc4122#section-4.4
  bytes[6] = (bytes[6] & 0x0f) | 0x40;
  bytes[8] = (bytes[8] & 0x3f) | 0x80;

  // Encode as UUIDv4
  final s = hex.encode(bytes).substring;
  return '${s(0, 8)}-${s(8, 12)}-${s(12, 16)}-${s(16, 20)}-${s(20)}';
```

Maybe, using `toRadixString` is nicer, since it avoids a dependency on `package:convert`.